bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -61,22 +61,35 @@ spec:
         - /bin/bash
         - -c
         - |
-          set -xe
+          set -xem
           if [[ -f /env/_master ]]; then
             set -o allexport
             source /env/_master
             set +o allexport
           fi
 
+          # We need to clean up on SIGTERM, as in some cases (upgrade+reboot)
+          # our preStop doesn't get called. When OpenShift graceful reboot
+          # is fixed (planned for 4.9+) this should be moved to preStop.
+          function quit {
+            echo "$(date -Iseconds) - stopping ovn-northd"
+            OVN_MANAGE_OVSDB=no /usr/share/ovn/scripts/ovn-ctl stop_northd
+            echo "$(date -Iseconds) - ovn-northd stopped"
+            exit 0
+          }
+          trap quit SIGTERM
+
           echo "$(date -Iseconds) - starting ovn-northd"
-          exec ovn-northd \
+          ovn-northd \
```

Comment:
`set -m` and `exec ...` was what worked in my testing

Reply:
mind you, so did `setsid` without flags but ymmv

Reply:
@dave-tucker if I exec then ovn-northd replaces the bash process and the SIGTERM won't trigger, because there is no more bash.